document-store/src/main/java/org/hypertrace/core/documentstore/metric/BaseDocStoreMetricProviderImpl.java

```diff
@@ -55,63 +58,70 @@ public List<DocStoreMetric> getCustomMetrics(final CustomMetricConfig customMetr
               });
 
       final Collection collection = dataStore.getCollection(collectionName);
-      final CloseableIterator<Document> iterator = collection.aggregate(query);
-
       final List<DocStoreMetric> metrics = new ArrayList<>();
 
-      while (iterator.hasNext()) {
-        final Document document = iterator.next();
-        final JsonNode node;
+      try (final CloseableIterator<Document> iterator =
+          collection.query(
+              query,
+              QueryOptions.builder()
+                  .dataFreshness(NEAR_REALTIME_FRESHNESS)
+                  .queryTimeout(Duration.ofMinutes(20))
```

Comment:
Is there a way to expose queryTime out as config?

Reply:
Yes. It's also available at the "ConnectionConfig" level, which can be overridden here if required.